app/navigation/ExpensesNavigator.scala

```diff
@@ -138,7 +138,20 @@ class ExpensesNavigator @Inject() () {
             _ => journeys.routes.SectionCompletedStateController.onPageLoad(taxYear, businessId, ExpensesEntertainment.toString, NormalMode)
 
     case ConstructionIndustryAmountPage =>
-      _ => taxYear => businessId => _ => journeys.expenses.construction.routes.ConstructionIndustryCYAController.onPageLoad(taxYear, businessId)
+      userAnswers =>
+        taxYear =>
+          businessId =>
+            accountingType =>
+              userAnswers.get(DisallowableSubcontractorCostsPage, Some(businessId)) match {
+                case Some(DisallowableSubcontractorCosts.Yes) if accountingType.contains(Accrual) =>
+                  construction.routes.ConstructionIndustryDisallowableAmountController.onPageLoad(taxYear, businessId, NormalMode)
+                case Some(_) =>
```

Comment:
Can we be explicit with `Some(DisallowableSubcontractorCosts.No)` here?

Reply:
2nd case is for Yes or No, so will leave it as is